cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -153,12 +157,12 @@ void stop() throws SchedulerException {
 
     @Override
     public void schedule(Id.Program programId, ProgramType programType, Iterable<Schedule> schedules) {
-      Preconditions.checkNotNull(scheduler, "Scheduler not yet initialized");
+      checkInitialized();
       Preconditions.checkNotNull(schedules);
 
-      String key = getJobKey(programId, programType);
+      String jobKey = getJobKey(programId, programType).getName();
```

Comment:
This would be a problem, if we deploy same application across different namespace. not related to this PR. maybe we should only deploy adapters only to default namespace.